### PR TITLE
chore(processor): add test for record support

### LINKED ISSUE #31: Add test that prove that records are supported

### PR DESCRIPTION
**Pull request by JoranVanBelle**
fixes #31

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -30,4 +30,5 @@ jobs:
           distribution: 'temurin'
           cache: maven
       - name: Build with Maven
-        run: mvn -B -ntp verify --file pom.xml
\ No newline at end of file
+        if: ${{ matrix.java }} == 11
```
Comment (jonas-grgt) [inline .github/workflows/maven.yml:33]: Shouldn't this be run from 11 and onward? `>= 11`

**Review by jonas-grgt — CHANGES_REQUESTED** [review]
Maybe we should also include record tests for STEP_WISE and STRICT strategies?

File: .github/workflows/maven.yml (hunk 1 above)
Comment (JoranVanBelle) [reply .github/workflows/maven.yml:33]: Records were only introduced in java 14, which means that it doesn't compile when testing with java 11

Comment (jonas-grgt) [reply .github/workflows/maven.yml:33]: Correct 💯

**Review by jonas-grgt — APPROVED** [review]

**Pull request merged**